proposals/NNNN-never-codable.md

```diff
@@ -0,0 +1,60 @@
+# Conform `Never` to `Codable`
+ 
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Author: [Nate Cook](https://github.com/natecook1000)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/swift#64899](https://github.com/apple/swift/pull/64899)
+
+## Introduction
+
+Extend `Never` so that it conforms to the `Encodable` and `Decodable` protocols, together known as `Codable`.
+
+Forum discussion: [https://forums.swift.org/t/pitch-conform-never-to-codable/64056](https://forums.swift.org/t/pitch-conform-never-to-codable/64056)
+
+## Motivation
+
+Swift can synthesize `Codable` conformance for any type that has `Codable` members. Generic types often participate in this synthesized conformance by constraining their generic parameters, like this `Either` type:
+
+```swift
+enum Either<A, B> {
+    case left(A)
+    case right(B)
+}
+
+extension Either: Codable where A: Codable, B: Codable {}
+```
+
+In this way, `Either` instances where both generic parameters are `Codable` are `Codable` themselves, such as an `Either<Int, Double>`. However, since `Never` isn't `Codable`, using `Never` as one of the parameters blocks the conditional conformance, even though it would be perfectly fine to encode or decode a type like `Either<Int, Never>`.
+
+## Proposed solution
+
+The standard library should add `Encodable` and `Decodable` conformance to the `Never` type.
+
+## Detailed design
+
+The `Encodable` conformance is simple — since it's impossible to have a `Never` instance, the `encode(to:)` method can simply be empty.
+
+The `Decodable` protocol requires the `init(from:)` initializer, which clearly can't create a `Never` instance. The implementation throws a `DecodingError` if decoding is attempted.
```

Comment:
I think the point Joe made in [this post on the pitch thread](https://forums.swift.org/t/pitch-conform-never-to-codable/64056/10) would be a good one to include in the proposal text here, to explain why you're specifically throwing an error and not trapping.

Do you want to call out the specific instance of `DecodingError` that the implementation throws?

Reply:
Included both points, thank you!